flexeval/core/metric/llm_score.py

```diff
@@ -49,7 +49,18 @@ def summarize_evaluator_scores(
         if score is None or category_key is None:
             continue
         if category_key in task_inputs:
-            category2valid_scores[task_inputs[category_key]].append(score)
+            category = task_inputs[category_key]
+            if isinstance(category, list):
+                for _category in category:
+                    if not isinstance(_category, str):
+                        msg = f"category should be either str or list[str], but got list containing {type(category)}"
+                        raise TypeError(msg)
+                    category2valid_scores[_category].append(score)
+            elif isinstance(category, str):
+                category2valid_scores[category].append(score)
+            else:
+                msg = f"category should be either str or list[str], but got {type(category)}"
+                raise TypeError(msg)
```

Comment:
How about being more flexible with the type checking for the category data?

The category values could be an int or any other type that's easily cast to str. They might also come as a tuple or set, which would still be perfectly acceptable.

By relaxing the type checks and reducing branching, the code becomes simpler:

```suggestion
        if category_key in task_inputs:
            categories = task_inputs[category_key]
            if not isinstance(categories, (list, tuple, set)):
                categories = [categories]
            for category in categories:
                category2valid_scores[category].append(score)
```

Reply:
Thank you for the suggestion which improves both the utility and readability!
Definitely accepted.

Reply:
Thank you for the bug fix🙏